src/sqlfluff/core/parser/markers.py

```diff
@@ -91,6 +91,31 @@ def from_point(
             **kwargs,
         )
 
+    @classmethod
+    def from_points(
+        cls,
+        start_point_marker: "PositionMarker",
+        end_point_marker: "PositionMarker",
+    ):
+        """Construct a position marker from the section between two points."""
+        return cls(
```

Comment:
Should we add sanity checks that:
* The two slices don't overlap
* The end slice starts after the start slice ends

Looking over this file, other factory functions could use similar checks also.

Reply:
Good idea, but I think this needs to be a separate issue. Some of these assumptions aren't actually true. I think they *should* be but even in the vanilla code before this PR they're not. I'll make a separate issue.

Reply:
See #4183 4183